### PR TITLE
Onedrive or sharepoint Alert

### PR DESCRIPTION
**Pull request by roaaattalla**

### CODE REVIEW

File: cyences_app_for_splunk/default/data/ui/views/cs_o365_reports.xml [hunk 1]
```diff
@@ -1003,7 +1003,7 @@
         <fieldForLabel>ApplicationDisplayName</fieldForLabel>
         <fieldForValue>ApplicationDisplayName</fieldForValue>
         <search>
-          <query>`cs_o365` sourcetype="o365:management:activity" Workload IN (OneDrive, SharePoint) Operation IN ("SharingSet", "AccessRequestApproved", "SharingRestored", "AnonymousLinkCreated", "SharingInvitationCreated", "SharingInvitationUpdated", "AddedToSecureLink", "SharingPolicyChanged") NOT TargetUserOrGroupType IN ("Member", "SharePointGroup") | dedup ApplicationDisplayName | table ApplicationDisplayName</query>
+          <query>`cs_o365` sourcetype="o365:management:activity" Workload IN (OneDrive, SharePoint) Operation IN ("SharingSet", "AccessRequestApproved", "SharingRestored", "AnonymousLinkCreated", "SharingInvitationCreated", "SharingInvitationUpdated", "AddedToSecureLink", "SharingPolicyChanged"."SharingLinkCreated") NOT TargetUserOrGroupType IN ("Member", "SharePointGroup") | dedup ApplicationDisplayName | table ApplicationDisplayName</query>
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/data/ui/views/cs_o365_reports.xml:1006]: dot should be replaced with comma

File: cyences_app_for_splunk/default/data/ui/views/cs_o365_reports.xml [hunk 2]
```diff
@@ -1026,16 +1026,17 @@
       </input>
       <table>
         <search>
-          <query>`cs_o365` sourcetype="o365:management:activity" Workload IN (OneDrive, SharePoint) Operation IN ("SharingSet", "AccessRequestApproved", "SharingRestored", "AnonymousLinkCreated", "SharingInvitationCreated", "SharingInvitationUpdated", "AddedToSecureLink", "SharingPolicyChanged") NOT TargetUserOrGroupType IN ("Member", "SharePointGroup") $tkn_exclude_teams$ $tkn_application_name$
+          <query>`cs_o365` sourcetype="o365:management:activity" Workload IN (OneDrive, SharePoint) Operation IN ("SharingSet", "AccessRequestApproved", "SharingRestored", "AnonymousLinkCreated", "SharingInvitationCreated", "SharingInvitationUpdated", "AddedToSecureLink", "SharingPolicyChanged"."SharingLinkCreated") NOT TargetUserOrGroupType IN ("Member", "SharePointGroup") $tkn_exclude_teams$ $tkn_application_name$
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/data/ui/views/cs_o365_reports.xml:1029]: dot should be replaced with comma

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 3]
```diff
@@ -2801,22 +2801,23 @@ display.page.search.tab = statistics
 display.page.search.mode = fast
 request.ui_dispatch_app = cyences_app_for_splunk
 request.ui_dispatch_view = search
-search = `cs_o365` sourcetype="o365:management:activity" _index_earliest=-61m@m _index_latest=-1m@m Workload IN (OneDrive, SharePoint) Operation IN ("SharingSet", "AccessRequestApproved", "SharingRestored", "AnonymousLinkCreated", "SharingInvitationCreated", "SharingInvitationUpdated", "AddedToSecureLink", "SharingPolicyChanged") NOT TargetUserOrGroupType IN ("Member", "SharePointGroup") \
+search = `cs_o365` sourcetype="o365:management:activity" _index_earliest=-61m@m _index_latest=-1m@m Workload IN (OneDrive, SharePoint) Operation IN ("SharingSet", "AccessRequestApproved", "SharingRestored", "AnonymousLinkCreated", "SharingInvitationCreated", "SharingInvitationUpdated", "AddedToSecureLink", "SharingPolicyChanged"."SharingLinkCreated") NOT TargetUserOrGroupType IN ("Member", "SharePointGroup") \
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:2804]: dot should be replaced with comma

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 4]
```diff
@@ -2801,22 +2801,23 @@ display.page.search.tab = statistics
...
 | `cs_o365_sharing_file_with_external_user_filter`
 action.cyences_notable_event_action = 1
 action.cyences_notable_event_action.param.filter_macro_name = cs_o365_sharing_file_with_external_user_filter
-action.cyences_notable_event_action.contributing_events = `cs_o365` sourcetype="o365:management:activity" _index_earliest=-61m@m _index_latest=-1m@m Workload IN (OneDrive, SharePoint) Operation IN ("SharingSet", "AccessRequestApproved", "SharingRestored", "AnonymousLinkCreated", "SharingInvitationCreated", "SharingInvitationUpdated", "AddedToSecureLink", "SharingPolicyChanged") NOT TargetUserOrGroupType IN ("Member", "SharePointGroup")
+action.cyences_notable_event_action.contributing_events = `cs_o365` sourcetype="o365:management:activity" Workload IN (OneDrive, SharePoint) Operation IN ("SharingSet", "AccessRequestApproved", "SharingRestored", "AnonymousLinkCreated", "SharingInvitationCreated", "SharingInvitationUpdated", "AddedToSecureLink", "SharingPolicyChanged"."SharingLinkCreated") NOT TargetUserOrGroupType IN ("Member", "SharePointGroup")
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:2820]: dot should be replaced with comma

File: cyences_app_for_splunk/default/data/ui/views/cs_o365_reports.xml [hunk 5]
```diff
@@ -1086,12 +1087,12 @@
...
+| stats values(Operation) AS Operations, values(ObjectId) as ObjectId , values(SharingLinkScope) as SharingLinkScope, values(Permission) as Permission, values(ApplicationDisplayName) as ApplicationDisplayName, values(Source) as Source,values(EventData) as EventData, values(ItemType) as ItemType,values(EventSource) as EventSource latest(_time) as LastTime earliest(_time) as FirstTime, values(Country) as Country, values(City) as City, values(Region) as Region by SourceFileName, UserId 
 | rename UserId as UserAccessingFile
 | `cs_human_readable_time_format(FirstTime)`
 | `cs_human_readable_time_format(LastTime)`
 | `cs_user_privilege_mapping(User)`
-| table UserAccessingFile,is_privileged_user,SourceFileName,ObjectId,EventSource,ItemType,Operations,FirstTime,LastTime,ApplicationDisplayName,Source,Country,City,Region
+| table UserAccessingFile,is_privileged_user,SourceFileName,Permission,ObjectId,EventSource,ItemType,Operations,FirstTime,LastTime,ApplicationDisplayName,Source,Country,City,Region
```
Comment (hardikhdholariya) [inline cyences_app_for_splunk/default/data/ui/views/cs_o365_reports.xml:1095]: Add SharingLinkScope field

**Review by hardikhdholariya — APPROVED** [review]

**Pull request merged**